### PR TITLE
Add workerPool

### PR DESCRIPTION
**Pull request by willianbvsanches13**
## Problem
_What problem are you trying to solve?_
We didn't control event quantity execution

## Solution
_How did you solve the problem?_
I made a workerPool to control execution quantity at the same time

## Checklist before code review
- [ ] Circle CI checks passed?
- [ ] Your branch is updated(rebased) with master?
- [ ] Did you write tests for your task, including regression tests?

## Reviewer Checklist
- [ ] I checked if the task has the tests
- [ ] I checked if the task has no security breach
- [ ] I checked if the task doesn't have unoptimized queries

## Checklist after merge
- [ ] If it's a _Sentry_ issue, mark it as resolved.
- [ ] Confirm deploy pass
- [ ] Notify product-team

## Deploy Notes
_Notes regarding deployment of the contained body of work. These should note any
new dependencies, new scripts, etc._

**New environment variables**:
- `env var` : env var details

**New scripts**:
- `script` : script details

**New dependencies**:
- `dependency` : dependency details

**New dev dependencies**:
- `dependency` : dependency details

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -45,6 +45,7 @@ Set env vars and execute init function:
 
 ```golang
 func init() {
+	os.Setenv("WORKERS", "4")
```
Comment (efreesen) [inline README.md:48]: If we are implementing this var here we must implement it on every other implementation, I think we should force it to a number now and then plan better how this var would work so we can implement it on every version of the EventPeople lib.

Maybe we could user this as the default for now:

```Go
runtime.NumCPU()
```

File: README.md [hunk 2]
```diff
@@ -86,7 +87,7 @@ func main() {
 There are 3 main interfaces to use `EventPeople` on your project:
 
 -   Calling `EventPeople.TriggerEmitter(event []*EventPeople.Event)` inside your project;
--   Calling `EventPeople.ListenTo(eventName string)` inside your project;
+-   Calling `EventPeople.GetMessageWithCallback(eventName string)` inside your project;
```
Comment (efreesen) [inline README.md]: We mustn't change the documented public API.

File: README.md [hunk 3]
```diff
@@ -136,7 +137,7 @@ Other important aspect of event consumming is the result of the processing we pr
 -   `Fail:` should be called when an error ocurred processing the event and the message should be requeued;
 -   `Reject:` should be called whenever a message should be discarded without being processed.
 
-Given you want to consume a single event inside your project you can use the `EventPeople.ListenTo` method. It consumes a single event, given there are events available to be consumed with the given name pattern.
+Given you want to consume a single event inside your project you can use the `EventPeople.GetMessageWithCallback` method. It consumes a single event, given there are events available to be consumed with the given name pattern.
```
Comment (efreesen) [inline README.md]: Same here.

File: README.md [hunk 4]
```diff
@@ -150,7 +151,7 @@ func main() {
   var eventName = "payment.payments.pay"
   var once = make(chan int)
 
-  EventPeople.ListenTo(eventName, func (event EventPeople.Event, context EventPeople.BaseListener) {
+  EventPeople.GetMessageWithCallback(eventName, func (event EventPeople.Event, context EventPeople.BaseListener) {
```
Comment (efreesen) [inline README.md]: And here.

File: README.md [hunk 5]
```diff
@@ -177,7 +178,7 @@ var once = make(chan int)
 func main() {
   var eventName = "payment.payments.pay.all"
 
-	EventPeople.ListenTo(eventName, func(event EventPeople.Event, context EventPeople.BaseListener) {
+	EventPeople.GetMessageWithCallback(eventName, func(event EventPeople.Event, context EventPeople.BaseListener) {
```
Comment (efreesen) [inline README.md]: And here.

**Review by efreesen — CHANGES_REQUESTED** [review]

File: examples/listener.go [hunk 6]
```diff
@@ -30,19 +30,19 @@ func listenerCallback(event EventPeople.Event, context EventPeople.ContextInterf
 var twice = make(chan int)
 
 func RunListener() {
-	EventPeople.ListenTo("resource.custom.*", listenerCallback)
-	EventPeople.ListenTo("resource.origin.*", listenerCallback)
+	EventPeople.GetMessageWithCallback("resource.custom.*", listenerCallback)
```
Comment (Badaueba) [inline examples/listener.go]: thinking about "GetMessageWithCallback" it's clear method name, it describes what it does, but i don't know if it follows other event_people projects standards

File: README.md (hunk 1 above)
Comment (willianbvsanches13) [reply README.md:48]: I'm using this if env is not setted

File: examples/daemon.go [hunk 7]
```diff
@@ -26,40 +29,35 @@ type PrivateMessageDaemon struct {
...
-func (custom *CustomEventListener) receive(event EventPeople.Event) {
+func receive(event EventPeople.Event, custom EventPeople.ContextInterface) {
 	var bodyDaemon = BodyStructureDaemon{}
 	event.SetStructBody(&bodyDaemon)
 
 	if bodyDaemon.Amount < 500 {
-		fmt.Println(fmt.Sprintf("[consumer] Got SKIPPED message:\n%d from %s ~> %s", bodyDaemon.Amount, bodyDaemon.Name, event.GetEventName()))
-		custom.Reject()
+		fmt.Println(fmt.Sprintf("%v : [consumer] Got SKIPPED message:\n%d from %s ~> %s", time.Now().Format("2006-01-02 15:04:05"), bodyDaemon.Amount, bodyDaemon.Name, event.GetEventName()))
+		custom.Fail()
```
Comment (efreesen) [inline examples/daemon.go]: Why `Fail`, this message is an example of the `Reject` method.

File: README.md (hunk 2 above)
Comment (willianbvsanches13) [reply README.md]: Fixed

File: README.md (hunk 3 above)
Comment (willianbvsanches13) [reply README.md]: Fixed

File: README.md (hunk 4 above)
Comment (willianbvsanches13) [reply README.md]: Fixed

File: README.md (hunk 5 above)
Comment (willianbvsanches13) [reply README.md]: Fixed

File: examples/listener.go (hunk 6 above)
Comment (willianbvsanches13) [reply examples/listener.go]: Fixed

**Review by willianbvsanches13 — COMMENTED** [review]
Fixed

**Review by efreesen — DISMISSED** [review]

**Review by efreesen — APPROVED** [review]

**Pull request merged**